tensorflow/contrib/tensorrt/convert/convert_graph.h

```diff
@@ -0,0 +1,34 @@
+/* Copyright 2017 The TensorFlow Authors. All Rights Reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+==============================================================================*/
+#ifndef TENSORFLOW_CONTRIB_TENSORRT_CONVERT_CONVERT_GRAPH_H_
+#define TENSORFLOW_CONTRIB_TENSORRT_CONVERT_CONVERT_GRAPH_H_
+
+#include <string>
+#include <vector>
+
+#include "tensorflow/core/framework/graph.pb.h"
+#include "tensorflow/core/lib/core/status.h"
+
+namespace tensorrt {
+namespace convert {
+
+tensorflow::Status ConvertGraphDefToTensorRT(
+    const tensorflow::GraphDef& graph_def,
+    const std::vector<std::string>& output_names, size_t max_batch_size,
+    size_t max_workspace_size, tensorflow::GraphDef* new_graph_def);
```

Comment:
Comment on `max_workspace_size`. Readers might not know what it means or even the units (in bytes or megabytes)?

Reply:
Also, who manages the work space? TF typically reserves all GPU memory ahead of time. TensorRT will run into OOM if it tries to allocate work space using cudaMalloc. @zheng-xq

Reply:
As of now, TensorRT doesn't have a API to use external allocator but they are aware of the situation and will try to push the api as soon as possible. Once it is available, we will use TF allocators to allocate memory. Until then, users will have to set a different allocation strategy, for example incremental allocation or fixed percentage allocation.

Reply:
That's very unfortunate. Either we should add an external allocator, or have caller to pass in the scratch space, similar to cudnn APIs

Reply:
We will have a means to use TF allocators directly and TensorRT team is working on it with high priority, but right now TensorRT doesn't have any means to let users control memory other than the size of the buffer. As soon as TensorRT API becomes ready we will make another PR for integrated memory allocation.